From f4a3948d9fdf0b68e7c66c27040bbf691205fa80 Mon Sep 17 00:00:00 2001
From: Michael Haubenwallner <michael.haubenwallner@ssi-schaefer.com>
Date: Tue, 21 May 2019 18:29:15 +0200
Subject: [PATCH 10/19] if already set, let CFLAGS/CXXFLAGS alone by default

Do not tweak CFLAGS/CXXFLAGS if already set and optimize or debug is not
explicitly requested by configure arguments.
---
 configure.in | 32 +++++++++++++++++++-------------
 1 file changed, 19 insertions(+), 13 deletions(-)

diff --git a/configure.in b/configure.in
index 20112bb..6cdadb4 100644
--- a/configure.in
+++ b/configure.in
@@ -112,12 +112,12 @@ AC_ARG_ENABLE(repo,
  use_repo=$enableval, use_repo=no)
 
 AC_ARG_ENABLE(optimize,
- [  --disable-optimize      do not use -O2 to compile],
- use_opt=$enableval, use_opt=yes)
+ [  --enable-optimize       use -O2 to compile],
+ use_opt=$enableval, ${CFLAGS+true} false && use_opt=default || use_opt=yes)
 
 AC_ARG_ENABLE(debug,
  [  --enable-debug          use -g to compile],
- use_debug=$enableval, use_debug=no)
+ use_debug=$enableval, ${CFLAGS+true} false && use_debug=default || use_opt=yes)
 
 AC_ARG_ENABLE(shared,
  [  --disable-shared        don't build shared libs],
@@ -1003,7 +1003,7 @@ else
       CONF_SHARED_CC="$CXX -shared"
       CONF_PICFLAGS=-fPIC
       CONF_LDSOFLAGS=-fPIC
-      if test X"$use_opt" != Xyes; then
+      if test X"$use_opt" = Xno; then
         # by default Intel uses -O so we need to disable it
         CONF_OPT_FLAGS=-O0
         use_opt=yes
@@ -1059,17 +1059,23 @@ fi
 #
 
 changequote(<<, >>)
-CXXFLAGS=`echo " $CXXFLAGS " | sed -e 's/ -g / /g' -e 's/ -O[0-9]* / /g'`
-CFLAGS=`echo " $CFLAGS " | sed -e 's/ -g / /g' -e 's/ -O[0-9]* / /g'`
-changequote([, ])
-if test X"$use_opt" != Xno; then
-  CXXFLAGS="$CONF_OPT_FLAGS $CXXFLAGS"
-  CFLAGS="$CONF_OPT_FLAGS $CFLAGS"
+if test X"$use_opt" != Xdefault; then
+  CXXFLAGS=`echo " $CXXFLAGS " | sed -e 's/ -O[0-9]* / /g'`
+  CFLAGS=`echo " $CFLAGS " | sed -e 's/ -O[0-9]* / /g'`
+  if test X"$use_opt" = Xyes; then
+    CXXFLAGS="$CONF_OPT_FLAGS $CXXFLAGS"
+    CFLAGS="$CONF_OPT_FLAGS $CFLAGS"
+  fi
 fi
-if test X"$use_debug" != Xno; then
-  CXXFLAGS="$CONF_DEBUG_FLAGS $CXXFLAGS"
-  CFLAGS="$CONF_DEBUG_FLAGS $CFLAGS"
+if test X"$use_debug" != Xdefault; then
+  CXXFLAGS=`echo " $CXXFLAGS " | sed -e 's/ -g / /g'`
+  CFLAGS=`echo " $CFLAGS " | sed -e 's/ -g / /g'`
+  if test X"$use_debug" = Xyes; then
+    CXXFLAGS="$CONF_DEBUG_FLAGS $CXXFLAGS"
+    CFLAGS="$CONF_DEBUG_FLAGS $CFLAGS"
+  fi
 fi
+changequote([, ])
 
 if test "x$use_memcheck" = "xyes" ; then
     case $target in
-- 
2.19.2

